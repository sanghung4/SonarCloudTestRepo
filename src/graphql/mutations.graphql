mutation purgeMincronCounts ($input: CountsInput!) {
  purgeMincronCounts (input: $input)
}

mutation addUser($userEmail: String!, $branchId: String!) {
  addUser(userEmail: $userEmail, branchId: $branchId) {
    success
    error {
      code
      message
        details
    }
  }
}

mutation removeUserBranch($userEmail: String!, $branchId: String!) {
  removeUserBranch(userEmail: $userEmail, branchId: $branchId) {
    success
    error {
      code
      message
      details
    }
  }
}

mutation updateUserEmail($oldUserEmail: String!, $newUserEmail: String!) {
  updateUserEmail(oldUserEmail: $oldUserEmail, newUserEmail: $newUserEmail) {
    success
    error {
      code
      message
      details
    }
  }
}

mutation deleteUser($userEmail: String!) {
  deleteUser(userEmail: $userEmail) {
    success
    error {
      code
      message
      details
    }
  }
}

mutation SetSpecialPrices($input: PriceChangeInput!) {
  specialPrices(input: $input) {
    successfulUploads {
      uploadedPath
      uploadedName
    }
  }
}

mutation LoadCount($branchId: String!, $countId: String!) {
  loadCount(branchId: $branchId, countId: $countId) {
    id
    branchId
    countId
    branchName
    status
    errorMessage
    createdAt
    totalProducts
  }
}

mutation removeBranchCounts($input: RemoveBranchCountsInput!){
  removeBranchCounts(input: $input){
    deletedCounts
    {
      countLocations
      countLocationItems
      varianceCountLocationItemQuantities
      countLocationItemQuantities
      writeIns
    }
  }
}