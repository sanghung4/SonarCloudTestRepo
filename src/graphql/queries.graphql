query GetCounts($input: CountsInput!) {
  counts(input: $input) {
    branchId
    branchName
    countId
    status
    createdAt
    totalProducts
    errorMessage
  }
}

query GetCountStatus($id: ID!) {
  countStatus(id: $id) {
    id
    branchId
    branchName
    countId
    status
    createdAt
    totalProducts
    errorMessage
  }
}

query GetPercentageTotal($input: MetricsSingleRangeInput!) {
  percentageTotal(input: $input) {
    response {
      division
      userCount
      userPercentage
      loginCount
      loginPercentage
      branchCount
      branchPercentage
    }
  }
}

query GetPercentageChange($input: MetricsDoubleRangeInput!) {
  percentageChange(input: $input) {
    response {
      division
      userChange
      loginChange
      branchChange
    }
  }
}

query GetTotalUsage($input: MetricsSingleRangeInput!) {
  totalUsage(input: $input) {
    response {
      division
      userCount
      loginCount
      branchCount
      branches {
        id
        city
        state
        userCount
        loginCount
      }
    }
  }
}

query GetTotalOverview($input: MetricsDoubleRangeInput!) {
  totalOverview(input: $input) {
    response {
      metric
      quantity
      percentageChange
    }
  }
}

query GetPaginatedSpecialPrices($input: SpecialPriceInput!, $pagingContext: PagingContext!) {
  paginatedSpecialPrices(input: $input, pagingContext: $pagingContext) {
    meta {
      page
      nextPage
      pageCount
      resultCount
      pageSize
      orderBy
      orderDirection
      productId
      customerId
    }
    results {
      productId
      customerId
      branch
      displayName
      customerDisplayName
      imageUrl
      manufacturer
      displayName
      manufacturerReferenceNumber
      salesperson
      priceLine
      customerSalesQuantity
      territory
      prices {
        type
        value
        currency
        displayName
      }
    }
  }
}

query GetPriceLines($input: SpecialPriceInput!) {
  priceLines(input: $input)
}

query SearchProductsEclipse($input: ProductSearchEclipseInput!) {
  searchProductsEclipse(input: $input) {
    pagination {
      currentPage
      totalItemCount
    }
    products {
      manufacturerName
      id
      name
      upc
      productNumber
      imageUrls {
        thumb
      }
    }
  }
}

query GetProductPrices($input: ProductPriceInput!) {
  productPrices(input: $input) {
    productId
    erpBranchNum
    cmp
    stdCost
    uom
    rateCardPrice
    rateCardName
    matchedBranch
    matchedClass
    matchedGroup
    matrixId
    customerId
    correlationId
  }
}

query GetCustomerById($input: String!) {
  customer(customerId: $input) {
    id
    homeBranch
    name
    isBillTo
    isShipTo
    addressLine1
    addressLine2
    city
    state
    postalCode
    countryCode
  }
}

query searchProductsKourier($input: 
  ProductSearchKourierInput!) {
  searchProductsKourier(input: $input) {
    prodSearch {
      errorCode
      errorMessage
      productIdCount
      products {
        productId
        displayField
      }
    }
  }
}

query branchUsersList {  
  branchUsersList {   
    branchId
    assignedUsers
  }
}

query getUserBranch($input: String!){
  userBranch(username: $input ){
    username
    branch
  }
}