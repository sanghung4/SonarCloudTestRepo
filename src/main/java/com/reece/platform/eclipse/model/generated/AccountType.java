/*
 * Eclipse API Developer Documentation
 * This documentation provides a list of API endpoints provided in this release as well as examples for using the various API endpoints
 *
 * OpenAPI spec version: 9.1.1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

package com.reece.platform.eclipse.model.generated;

import java.util.Objects;
import java.util.Arrays;
import io.swagger.v3.oas.annotations.media.Schema;
import javax.validation.constraints.*;
import javax.validation.Valid;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 *  List of valid Account types which may be entered in the \&quot;AccountType\&quot; field.  &#x27;1 - Liability&#x27;  &#x27;2 - Asset&#x27;
 */
public enum AccountType {
  LIABILITY("Liability"),
  ASSET("Asset");

  private String value;

  AccountType(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static AccountType fromValue(String text) {
    for (AccountType b : AccountType.values()) {
      if (String.valueOf(b.value).equals(text)) {
        return b;
      }
    }
    return null;
  }
}
