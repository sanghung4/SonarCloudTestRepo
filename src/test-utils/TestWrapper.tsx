import React, { JSXElementConstructor, ReactElement, ReactNode } from 'react';

import { ThemeProvider } from '@dialexa/reece-component-library';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { OktaAuth } from '@okta/okta-auth-js';
import { render as jestRender, RenderResult } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { MemoryRouter } from 'react-router-dom';

import { UserContextType, AuthContext } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';
import i18n from 'test-utils/mockI18N';
import {
  SelectedAccountsContext,
  SelectedAccountsContextType
} from 'providers/SelectedAccountsProvider';
import { CartContext, CartContextType } from 'providers/CartProvider';
import {
  CheckoutContext,
  CheckoutContextType,
  initialCheckoutContextValues
} from 'Checkout/CheckoutProvider';

/**
 * NOTE:
 * To setup mocks follow the examples here:
 * https://www.apollographql.com/docs/react/development-testing/testing/
 *
 * To get a successful response in your test (not just the loading state)
 * you can do the following:
 *
 * it('Should render', async () => {
 *   const { container } = renderMockRouter(<MyComponent />, mocks);
 *
 *   await waitFor(() => new Promise((res) => setTimeout(res, 0)));
 *
 *   expect(container).toMatchSnapshot();
 * });
 *
 */

/******************************/
/* Types                      */
/******************************/
type Props = {
  children?: ReactNode;
  authConfig?: Partial<UserContextType>;
  selectedAccountsConfig?: Partial<SelectedAccountsContextType>;
  cartConfig?: Partial<CartContextType>;
  checkoutConfig?: Partial<CheckoutContextType>;
  mocks?: ReadonlyArray<MockedResponse>;
};

type WrapperProps<TKey extends keyof Props> = {
  value: Props[TKey];
  children: React.ReactNode;
};

/******************************/
/* Defaults                   */
/******************************/
const defaultAuth: UserContextType = {
  authState: {
    isAuthenticated: true
  },
  isLoggingOut: false,
  oktaAuth: new OktaAuth({ issuer: 'http://test.com' }),
  profile: {
    userId: 'testuser',
    permissions: Object.values(Permission),
    isEmployee: false,
    isVerified: false
  }
};

const defaultSelectedAccounts: SelectedAccountsContextType = {
  clearAccounts: () => {},
  updateAccounts: () => {},
  updateShippingBranchId: () => {},
  loading: false,
  selectedAccounts: {
    billTo: { erpAccountId: '' },
    shipTo: { erpAccountId: '' },
    shippingBranchId: ''
  },
  isMincron: false,
  isEclipse: false
};

const defaultCartContextValue: CartContextType = {
  addItemToCart: () => {},
  addItemsToCart: () => {},
  addAllListItemsToCart: () => {},
  cart: undefined,
  cartLoading: false,
  checkingOutWithQuote: false,
  contractBranch: undefined,
  clearContract: () => {},
  clearQuote: () => {},
  deleteCartItems: () => new Promise(() => {}),
  disableAddToCart: false,
  deleteItem: () => {},
  getUserCart: () => new Promise(() => {}),
  itemCount: 0,
  itemLoading: undefined,
  lineNotes: {},
  releaseContractToCart: () => {},
  setContract: () => {},
  setLineNotes: () => {},
  setQuoteId: () => {},
  setQuoteData: () => {},
  setQuoteShipToId: () => {},
  setPreviousCart: () => {},
  setSelectedBranch: () => {},
  updateCart: undefined,
  updateDelivery: undefined,
  updateItemQuantity: () => {},
  updateWillCall: undefined,
  updateWillCallBranch: undefined,
  refreshCart: undefined,
  isWillCall: false,
  deleteCreditCardFromCart: () => new Promise(() => {})
};

/******************************/
/* Wrappers                   */
/******************************/
function SelectedAccountsProviderWrapper({
  value,
  children
}: WrapperProps<'selectedAccountsConfig'>) {
  if (value) {
    return (
      <SelectedAccountsContext.Provider
        value={{ ...defaultSelectedAccounts, ...value }}
      >
        {children}
      </SelectedAccountsContext.Provider>
    );
  }

  return <>{children}</>;
}

function CartProviderWrapper({ value, children }: WrapperProps<'cartConfig'>) {
  if (value) {
    return (
      <CartContext.Provider value={{ ...defaultCartContextValue, ...value }}>
        {children}
      </CartContext.Provider>
    );
  }

  return <>{children}</>;
}

function CheckoutProviderWrapper({
  value,
  children
}: WrapperProps<'checkoutConfig'>) {
  if (value) {
    return (
      <CheckoutContext.Provider
        value={{ ...initialCheckoutContextValues, ...value }}
      >
        {children}
      </CheckoutContext.Provider>
    );
  }

  return <>{children}</>;
}

function TestWrapper(props: Props) {
  return (
    <ThemeProvider>
      <I18nextProvider i18n={i18n}>
        <AuthContext.Provider value={{ ...defaultAuth, ...props.authConfig }}>
          <MockedProvider mocks={props.mocks ?? []}>
            <SelectedAccountsProviderWrapper
              value={props.selectedAccountsConfig}
            >
              <CartProviderWrapper value={props.cartConfig}>
                <CheckoutProviderWrapper value={props.checkoutConfig}>
                  {props.children}
                </CheckoutProviderWrapper>
              </CartProviderWrapper>
            </SelectedAccountsProviderWrapper>
          </MockedProvider>
        </AuthContext.Provider>
      </I18nextProvider>
    </ThemeProvider>
  );
}

function jestRenderFn(wrapped: JSX.Element) {
  return jestRender(wrapped, {
    wrapper: MemoryRouter as
      | JSXElementConstructor<{
          children: ReactElement<any, string | JSXElementConstructor<any>>;
        }>
      | undefined
  });
}

export function render(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return jestRenderFn(wrapped);
}
export type TestRender = RenderResult<
  typeof import('@testing-library/dom/types/queries'),
  Element | DocumentFragment
>;

export function renderWithComponent(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return {
    render: jestRenderFn(wrapped),
    component: wrapped
  };
}

export default TestWrapper;
