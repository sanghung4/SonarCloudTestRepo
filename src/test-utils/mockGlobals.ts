export function mockGoogle() {
  global.window.google = {
    maps: {
      // @ts-ignore
      places: {
        Autocomplete: jest.fn().mockImplementation(() => ({
          setFields: jest.fn(),
          setComponentRestrictions: jest.fn(),
          addListener: jest.fn()
        }))
      },
      Size: jest.fn(),
      Point: jest.fn()
    }
  };
}

export function mockGeolocation(lat: number, lng: number, err?: boolean) {
  // @ts-ignore
  global.navigator.geolocation = {
    getCurrentPosition: jest.fn().mockImplementation((onSuccess, onError) => {
      onSuccess({
        coords: {
          latitude: lat ?? 0,
          longitude: lng ?? 0
        }
      });
      if (err) {
        onError(new Error());
      }
    }),
    watchPosition: jest.fn().mockImplementation((onSuccess, onError) => {
      onSuccess({
        coords: {
          latitude: lat ?? 0,
          longitude: lng ?? 0
        }
      });
      if (err) {
        onError(new Error());
      }
    })
  };
}

export function mockLocalStorage() {
  // @ts-ignore
  global.window.localStorage = class mockLocalStorage {
    store: { [key: string]: string };
    constructor() {
      this.store = {};
    }
    clear() {
      this.store = {};
    }
    get length() {
      return Object.keys(this.store).length;
    }
    getItem(key: string) {
      return this.store[key] || null;
    }
    key(i: number) {
      return Object.keys(this.store)[i] || null;
    }
    removeItem(key: string) {
      delete this.store[key];
    }
    setItem(key: string, value: string) {
      this.store[key] = String(value);
    }
  };
}
