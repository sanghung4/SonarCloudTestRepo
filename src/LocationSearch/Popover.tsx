import React, { useMemo } from 'react';

import { Box, Grid, Link, Typography } from '@dialexa/reece-component-library';
import { InfoWindow } from '@react-google-maps/api';
import { useTranslation } from 'react-i18next';

import { Branch } from 'generated/graphql';
import { COMING_SOON } from 'LocationSearch';

type Props = {
  branch: Branch;
  position: google.maps.LatLng;
  onCloseClick: () => void;
};

function Popover(props: Props) {
  /**
   * Custom Hooks
   */
  const { t } = useTranslation();

  /**
   * Memos
   */
  const directionsUrl = useMemo(directionsUrlMemo, [props.branch]);
  const address = useMemo(addressMemo, [
    props.branch.address1,
    props.branch.address2,
    props.branch.city,
    props.branch.state,
    props.branch.zip
  ]);

  return (
    <InfoWindow
      position={{
        lat: props.position.lat() ?? 0,
        lng: props.position.lng() ?? 0
      }}
      onCloseClick={props.onCloseClick}
      options={{
        pixelOffset: new google.maps.Size(0, -15)
      }}
    >
      <Box
        overflow="hidden"
        py={1}
        px={2}
        data-testid={`google-maps-box-${props?.branch?.branchId}`}
      >
        <Grid container spacing={1} direction="column">
          <Grid item>
            <Typography
              variant="h5"
              color="primary"
              data-testid={`google-maps-branch-name-${props?.branch?.branchId}`}
            >
              {props.branch.name}
            </Typography>
          </Grid>
          <Grid item>
            <Typography
              variant="body2"
              data-testid={`google-maps-branch-address-${props?.branch?.branchId}`}
            >
              {address}
            </Typography>
          </Grid>
          <Grid item container spacing={1}>
            <Grid item>
              <Typography variant="body2">{t('common.hours')}:</Typography>
            </Grid>
            <Grid item data-testid={`google-maps-branch-hours`}>
              {props.branch.businessHours?.split('; ').map((h, i) => (
                <Typography
                  key={i}
                  color={h === COMING_SOON ? 'success' : 'inherit'}
                  variant="body2"
                >
                  {h.replace(':', '')}
                </Typography>
              )) ?? (
                <Typography variant="body2">
                  {t('locationSearch.noHours')}
                </Typography>
              )}
            </Grid>
          </Grid>
          <Grid item>
            <Link
              href={`tel:${props.branch.phone}`}
              sx={{
                color: 'primary02.main',
                fontWeight: 500
              }}
              data-testid={`google-maps-branch-phone-${props?.branch?.branchId}`}
            >
              {props.branch.phone}
            </Link>
          </Grid>
          <Grid item container alignItems="flex-end">
            <Grid item xs>
              <Link
                href={directionsUrl}
                target="_blank"
                rel="noreferer"
                sx={{ color: 'primary02.main', pt: 1.5 }}
                data-testid={`google-maps-branch-directions-${props?.branch?.branchId}`}
              >
                {t('locationSearch.directions')}
              </Link>
            </Grid>
            <Grid
              item
              data-testid={`google-maps-branch-distance-${props?.branch?.branchId}`}
            >
              {`${props.branch?.distance?.toFixed(1)} ${t('common.miles')}`}
            </Grid>
          </Grid>
        </Grid>
      </Box>
    </InfoWindow>
  );

  /**
   * Memo defs
   */

  function directionsUrlMemo() {
    let address = `${props.branch.address1} ${props.branch.address2 ?? ''} ${
      props.branch.city
    } ${props.branch.state} ${props.branch.zip}`;
    return `https://www.google.com/maps/dir/?api=1&destination=${encodeURIComponent(
      address
    )}`;
  }
  function addressMemo() {
    const secondaryAddress = props.branch.address2
      ? ` ${props.branch.address2}`
      : '';
    const cityStateZip = `, ${props.branch.city}, ${props.branch.state} ${props.branch.zip}`;
    return `${props.branch.address1}${secondaryAddress}${cityStateZip}`;
  }
}

export default Popover;
