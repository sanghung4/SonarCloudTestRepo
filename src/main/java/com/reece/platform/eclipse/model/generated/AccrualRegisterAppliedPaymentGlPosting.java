/*
 * Eclipse API Developer Documentation
 * This documentation provides a list of API endpoints provided in this release as well as examples for using the various API endpoints
 *
 * OpenAPI spec version: 9.1.1
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

package com.reece.platform.eclipse.model.generated;

import java.util.Objects;
import java.util.Arrays;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import com.reece.platform.eclipse.model.generated.Money;
import io.swagger.v3.oas.annotations.media.Schema;
import javax.validation.constraints.*;
import javax.validation.Valid;
/**
 * Object representing an Eclipse AccrualRegisterAppliedPaymentGlPosting.
 */
@Schema(description = "Object representing an Eclipse AccrualRegisterAppliedPaymentGlPosting.")

public class AccrualRegisterAppliedPaymentGlPosting {
  @JsonProperty("glBranch")
  private String glBranch = null;

  @JsonProperty("glAccount")
  private Integer glAccount = null;

  @JsonProperty("glAmount")
  private Money glAmount = null;

  @JsonProperty("glAutopostCode")
  private String glAutopostCode = null;

  public AccrualRegisterAppliedPaymentGlPosting glBranch(String glBranch) {
    this.glBranch = glBranch;
    return this;
  }

   /**
   * Desc: The AccrualRegisterAppliedPaymentGlPosting GlBranch  File: AR  Attr: 4,x,y
   * @return glBranch
  **/
  @Schema(description = "Desc: The AccrualRegisterAppliedPaymentGlPosting GlBranch  File: AR  Attr: 4,x,y")
  public String getGlBranch() {
    return glBranch;
  }

  public void setGlBranch(String glBranch) {
    this.glBranch = glBranch;
  }

  public AccrualRegisterAppliedPaymentGlPosting glAccount(Integer glAccount) {
    this.glAccount = glAccount;
    return this;
  }

   /**
   * Desc: The AccrualRegisterAppliedPaymentGlPosting GlAccount  File: AR  Attr: 4,x,y
   * @return glAccount
  **/
  @Schema(description = "Desc: The AccrualRegisterAppliedPaymentGlPosting GlAccount  File: AR  Attr: 4,x,y")
  public Integer getGlAccount() {
    return glAccount;
  }

  public void setGlAccount(Integer glAccount) {
    this.glAccount = glAccount;
  }

  public AccrualRegisterAppliedPaymentGlPosting glAmount(Money glAmount) {
    this.glAmount = glAmount;
    return this;
  }

   /**
   * Get glAmount
   * @return glAmount
  **/
  @Valid
  @Schema(description = "")
  public Money getGlAmount() {
    return glAmount;
  }

  public void setGlAmount(Money glAmount) {
    this.glAmount = glAmount;
  }

  public AccrualRegisterAppliedPaymentGlPosting glAutopostCode(String glAutopostCode) {
    this.glAutopostCode = glAutopostCode;
    return this;
  }

   /**
   * Desc: The AccrualRegisterAppliedPaymentGlPosting GlAutopostCode  File: AR  Attr: 34,x,y
   * @return glAutopostCode
  **/
  @Schema(description = "Desc: The AccrualRegisterAppliedPaymentGlPosting GlAutopostCode  File: AR  Attr: 34,x,y")
  public String getGlAutopostCode() {
    return glAutopostCode;
  }

  public void setGlAutopostCode(String glAutopostCode) {
    this.glAutopostCode = glAutopostCode;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    AccrualRegisterAppliedPaymentGlPosting accrualRegisterAppliedPaymentGlPosting = (AccrualRegisterAppliedPaymentGlPosting) o;
    return Objects.equals(this.glBranch, accrualRegisterAppliedPaymentGlPosting.glBranch) &&
        Objects.equals(this.glAccount, accrualRegisterAppliedPaymentGlPosting.glAccount) &&
        Objects.equals(this.glAmount, accrualRegisterAppliedPaymentGlPosting.glAmount) &&
        Objects.equals(this.glAutopostCode, accrualRegisterAppliedPaymentGlPosting.glAutopostCode);
  }

  @Override
  public int hashCode() {
    return Objects.hash(glBranch, glAccount, glAmount, glAutopostCode);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class AccrualRegisterAppliedPaymentGlPosting {\n");
    
    sb.append("    glBranch: ").append(toIndentedString(glBranch)).append("\n");
    sb.append("    glAccount: ").append(toIndentedString(glAccount)).append("\n");
    sb.append("    glAmount: ").append(toIndentedString(glAmount)).append("\n");
    sb.append("    glAutopostCode: ").append(toIndentedString(glAutopostCode)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}
