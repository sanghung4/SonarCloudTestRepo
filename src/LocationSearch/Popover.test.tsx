import { ReactNode } from 'react';

import { Branch } from 'generated/graphql';
import Popover from 'LocationSearch/Popover';
import { mockGoogle } from 'test-utils/mockGlobals';
import { render } from 'test-utils/TestWrapper';

const branch: Branch = {
  actingBranchManager: 'C.R. Kirby ',
  actingBranchManagerEmail: 'seth+reecebranchmanager@dialexa.com',
  actingBranchManagerPhone: '817-659-3953',
  address1: '15850 Dallas Parkway',
  address2: 'Suite 105',
  branchId: '1149',
  businessHours: 'M-F: 7:00am - 5:00pm',
  city: 'Dallas',
  distance: 1.7589941,
  entityId: '546983',
  isBandK: false,
  isHvac: false,
  isPlumbing: true,
  isWaterworks: false,
  latitude: 32.9643304,
  longitude: -96.8202783,
  name: 'Morrison Supply',
  phone: '(469) 680-3291',
  state: 'Texas',
  website: 'https://www.morrisonsupply.com',
  workdayId: null,
  workdayLocationName: null,
  zip: '75248',
  __typename: 'Branch'
};

const mockLatLng = {
  lat: () => 0,
  lng: () => 0,
  equals: jest.fn(),
  toJSON: jest.fn(),
  toUrlValue: jest.fn()
};

type EmptyCompProps = {
  children: ReactNode;
};

jest.mock('@react-google-maps/api', () => ({
  InfoWindow: ({ children }: EmptyCompProps) => <div>{children}</div>
}));

describe('LocationSearch Popover tests', () => {
  beforeEach(() => {
    mockGoogle();
  });

  it('should render correctly', () => {
    const { container } = render(
      <Popover branch={branch} position={mockLatLng} onCloseClick={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });
});
