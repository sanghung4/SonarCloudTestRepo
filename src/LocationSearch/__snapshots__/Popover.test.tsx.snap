// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationSearch Popover tests should render correctly 1`] = `
<div>
  <div>
    <div
      class="MuiBox-root css-1jbsqn1"
      data-testid="google-maps-box-1149"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column css-ffjoah-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-1bvxf5q-MuiTypography-root"
            data-testid="google-maps-branch-name-1149"
          >
            Morrison Supply
          </h5>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-rwfr69-MuiTypography-root"
            data-testid="google-maps-branch-address-1149"
          >
            15850 Dallas Parkway Suite 105, Dallas, Texas 75248
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 css-v8g4am-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-rwfr69-MuiTypography-root"
            >
              Hours
              :
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            data-testid="google-maps-branch-hours"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-nsm5pd-MuiTypography-root"
            >
              M-F 7:00am - 5:00pm
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-j9o9zr-MuiTypography-root-MuiLink-root"
            data-testid="google-maps-branch-phone-1149"
            href="tel:(469) 680-3291"
          >
            (469) 680-3291
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item css-10k7qhr-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1z0dvx8-MuiGrid-root"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-13if51i-MuiTypography-root-MuiLink-root"
              data-testid="google-maps-branch-directions-1149"
              href="https://www.google.com/maps/dir/?api=1&destination=15850%20Dallas%20Parkway%20Suite%20105%20Dallas%20Texas%2075248"
              rel="noreferer"
              target="_blank"
            >
              Directions
            </a>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            data-testid="google-maps-branch-distance-1149"
          >
            1.8 miles
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
